src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,856 @@
+                                                         additionalLocations: null,
+                                                         properties: null,
+                                                         arg1,
+                                                         arg2);
+                reportDiagnostic(diagnostic);
+            }
+
+            private static IEnumerable<INamedTypeSymbol> GetAttributesForMethodsToIgnore(Compilation compilation)
+            {
+                // Ignore conditional methods (One conditional will often call another conditional method as its only use of a parameter)
+                var conditionalAttribte = compilation.ConditionalAttribute();
+                if (conditionalAttribte != null)
+                {
+                    yield return conditionalAttribte;
+                }
+
+                // Ignore methods with special serialization attributes (All serialization methods need to take 'StreamingContext')
+                var onDeserializingAttribute = compilation.OnDeserializingAttribute();
+                if (onDeserializingAttribute != null)
+                {
+                    yield return onDeserializingAttribute;
+                }
+
+                var onDeserializedAttribute = compilation.OnDeserializedAttribute();
+                if (onDeserializedAttribute != null)
+                {
+                    yield return onDeserializedAttribute;
+                }
+
+                var onSerializingAttribute = compilation.OnSerializingAttribute();
+                if (onSerializingAttribute != null)
+                {
+                    yield return onSerializingAttribute;
+                }
+
+                var onSerializedAttribute = compilation.OnSerializedAttribute();
+                if (onSerializedAttribute != null)
+                {
+                    yield return onSerializedAttribute;
+                }
+
+                // Don't flag obsolete methods.
+                var obsoleteAttribute = compilation.ObsoleteAttribute();
+                if (obsoleteAttribute != null)
+                {
+                    yield return obsoleteAttribute;
+                }
+            }
+
+            private bool IsUnusedParameterCandidate(IParameterSymbol parameter)
+            {
+                // Ignore implicitly declared parameters and methods, extern methods, abstract methods,
+                // virtual methods, overrides, interface implementations, accessors and lambdas.
+                if (parameter.IsImplicitlyDeclared ||
+                    parameter.Name == "_" ||
+                    !(parameter.ContainingSymbol is IMethodSymbol method) ||
+                    method.IsImplicitlyDeclared ||
+                    method.IsExtern ||
+                    method.IsAbstract ||
+                    method.IsVirtual ||
+                    method.IsOverride ||
+                    !method.ExplicitOrImplicitInterfaceImplementations().IsEmpty ||
```

Comment:
it's worth commenting that this is a weak heuristic and will be wrong for implicit interface implementations across symbols.  We should comment when we're aware that something isn't relaly correct, but which we hope is good enogh.  it helps explain to the maintainer the difference between intentional and mistaken feature deficiencies.